### PR TITLE
bpo-31985: Deprecate aifc.openfp

### PR DESCRIPTION
**Pull request by briancurtin**
aifc.openfp had pointed to aifc.open since 1993* and it was both undocumented and untested since being assigned as a matter of backwards compatibility. This change begins its formal deprecation.

\* https://github.com/python/cpython/commit/7bc817d5ba917528e8bd07ec461c635291e7b06a

https://bugs.python.org/issue31985

### CODE REVIEW

File: Lib/test/test_aifc.py
Comment (berkerpeksag) [inline Lib/test/test_aifc.py]: Style nit: `assertIsInstance` might be better option here.

File: Lib/test/test_aifc.py
Comment (berkerpeksag) [inline Lib/test/test_aifc.py]: Style nit: I'd use `assertEqual` instead.

File: Lib/test/test_aifc.py
Comment (berkerpeksag) [inline Lib/test/test_aifc.py]: We could replace those two lines with `assertWarns`.

**Review by berkerpeksag — COMMENTED** [review]
LGTM, I just left some minor comments.

File: Lib/test/test_aifc.py
Comment (briancurtin) [reply Lib/test/test_aifc.py]: I initially thought that as well, but `category` isn't an instance there.

The other comments look good and I'll push a fix for them.

Comment (briancurtin) [reply Lib/test/test_aifc.py]: By using `assertWarns` we get the behavior we needed from those two `assertTrue` lines, so I just removed them in ebe685d.

**Comment by briancurtin** [comment]
Per https://bugs.python.org/issue31985#msg305995 don't merge this yet.

File: Lib/aifc.py
Comment (serhiy-storchaka) [inline Lib/aifc.py:918]: Please add a comment that `test_pyclbr.py` should be updated after removing `openfp`.

**Review by serhiy-storchaka — COMMENTED** [review]
Thank you Brian!

Please replace `wave.openfp` with `wave.open` in `sndhdr.py`. Add `versiondeprecated` in the documentation for `wave.openfp` and `sunau.openfp`.

File: Lib/aifc.py
Comment (briancurtin) [reply Lib/aifc.py:918]: Comment added. After this merges I'm going to create an issue in Roundup and assign it to myself so it gets formally tracked.

File: Doc/library/sunau.rst
Comment (serhiy-storchaka) [inline Doc/library/sunau.rst]: Oh, sorry, the correct name of this directive is `deprecated`! Or `deprecated-removed` if you have assigned the date of removing.

**Review by serhiy-storchaka — APPROVED** [review]

**Pull request merged**
